src/ansys/sound/core/signal_processing/filter.py

```diff
@@ -0,0 +1,311 @@
+            file shall have the same text format (with the header `AnsysSound_FRF`), as supported
+            by Ansys Sound SAS. If ``file`` is specified, parameters ``a_coefficients`` and
+            ``b_coefficients`` are ignored.
+        signal : Field
+            Mono signal to filter, as a DPF field.
+        """
+        super().__init__()
+
+        self.__operator_design = Operator(ID_OPERATOR_DESIGN)
+        self.__operator_load = Operator(ID_OPERATOR_LOAD)
+        self.__operator_filter = Operator(ID_OPERATOR_FILTER)
+
+        self.__sampling_frequency = sampling_frequency
+
+        if file != "":
+            if a_coefficients != [] or b_coefficients != []:
+                warnings.warn(
+                    PyAnsysSoundWarning(
+                        "Specified parameters a_coefficients and b_coefficients are ignored "
+                        "because FRF file is also specified."
+                    )
+                )
+            self.design_FIR_from_FRF_file(file)
+        else:
+            self.a_coefficients = a_coefficients
+            self.b_coefficients = b_coefficients
+
+        self.signal = signal
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if len(self.a_coefficients) > 5:
+            str_a = str(self.a_coefficients[:5])[:-1] + ", ... ]"
+        elif len(self.a_coefficients) == 0:
+            str_a = "Not set"
+        else:
+            str_a = str(self.a_coefficients)
+
+        if len(self.b_coefficients) > 5:
+            str_b = str(self.b_coefficients[:5])[:-1] + ", ... ]"
+        elif len(self.b_coefficients) == 0:
+            str_b = "Not set"
+        else:
+            str_b = str(self.b_coefficients)
+
+        return (
+            f"Sampling frequency: {self.__sampling_frequency:.1f} Hz\n"
+            f"Numerator coefficients (B): {str_b}\n"
+            f"Denominator coefficients (A): {str_a}"
+        )
+
+    @property
+    def a_coefficients(self) -> list[float]:
+        """Denominator coefficients of the filter's transfer function."""
+        return self.__a_coefficients
+
+    @a_coefficients.setter
+    def a_coefficients(self, coefficients: list[float]):
+        """Set filter's denominator coefficients."""
+        self.__a_coefficients = coefficients
+
+    @property
+    def b_coefficients(self) -> list[float]:
+        """Numerator coefficients of the filter's transfer function."""
+        return self.__b_coefficients
+
+    @b_coefficients.setter
+    def b_coefficients(self, coefficients: list[float]):
+        """Set filter's numerator coefficients."""
+        self.__b_coefficients = coefficients
+
+    @property
+    def signal(self) -> Field:
+        """Input signal as a DPF field."""
+        return self.__signal
+
+    @signal.setter
+    def signal(self, signal: Field):
+        """Set input signal."""
+        if signal is not None:
```

Comment:
@ansaminard why not using only one `if` statement ? 

`if signal is not None and not (isinstance(signal, Field)):`

Reply:
@a-bouth Because I am doing other tests below